packages/cloud_functions/cloud_functions/lib/src/https_callable.dart

```diff
@@ -30,21 +31,27 @@ class HttpsCallable {
   /// the user is also automatically included.
   Future<HttpsCallableResult<T>> call<T>([dynamic parameters]) async {
     _assertValidParameterType(parameters);
-    assert(_delegate != null);
-    return HttpsCallableResult<T>._(await _delegate.call(parameters));
+    return HttpsCallableResult<T>._(await delegate.call(parameters));
   }
 }
 
 /// Asserts whether a given call parameter is a valid type.
 void _assertValidParameterType(dynamic parameter, [bool isRoot = true]) {
   if (parameter is List) {
-    return parameter
-        .forEach((element) => _assertValidParameterType(element, false));
+    for (final element in parameter) {
```

Comment:
Those `for` aren't tree-shaked on release build.

Either we need to wrap the entire function inside a `if (kDebugMode)` or we need to use like like so:

```dart
assert(_validParameterType(parameter));
```

Reply:
Done